### ISSUE TITLE
Weight Conversion

### ISSUE THREAD

**Issue by Callidior** [qubvel/efficientnet#19]
Could you provide some details about how you converted the weights?

The official checkpoints for EfficientNet-B4 and -B5 have [recently been released][1] and I tried to convert them, but my B4 model only achieves 79.7% top-1 accuracy instead of 82.6%.

[1]: https://github.com/tensorflow/tpu/issues/377#issuecomment-502528921

**Comment by qubvel** [comment]
Hi @Callidior 
[Here](https://gist.github.com/qubvel/8a6d23e485ebb1611b330bbfa534a6ac) is a note about how to adapt weights for my models.
New weights are released #20

**Comment by Callidior** [comment]
Okay, I found my mistake. I was reading the checkpoint files directly using `tensorflow.python.pywrap_tensorflow.NewCheckpointReader` and used the final weights for the variables. In contrast, `tf.train.Saver.restore` uses the *exponential moving average* of the weights, which are also stored in the checkpoint. Using these, I achieve the reported accuracies.
